vcr/stubs/aiohttp_stubs/__init__.py

```diff
@@ -91,7 +91,7 @@ def play_responses(cassette, vcr_request):
 
     # If we're following redirects, continue playing until we reach
     # our final destination.
-    while 300 <= response.status <= 399:
+    while 300 <= response.status <= 399 and response.status != 304:
         next_url = URL(response.url).with_path(response.headers["location"])
```

Comment:
It seems like there are other status codes that also don't contain a location header (_e.g._ `300` and `308`).

Perhaps instead of excluding `304`s in the `while` loop, you could check for the existence of `response.headers["location"]` before setting `next_url` and `break` if not set?

Reply:
Yeah this way it covers all cases. I made the suggested change